### DISCUSSION TITLE
concurrency limit being ignored

### DISCUSSION THREAD

**Discussion by wyardley** [renovatebot/renovate#11508] [Request Help]
I have a new project that I've enabled Renovate GH app for. Have used Renovate quite a bit, but self-hosted in the past.

Originally, I only had `prconcurrentLimit` set, and had it set to a string, not an int. After I temporarily enabled a setting (may or may not be related), I noticed that we had a lot of pending builds from Renovate having branches open for each of its pending changes, not just the ones for which there were open PRs; in other words, `branchConcurrentLimit` didn't seem to be honoring the implicit relationship with `prConcurrentLimit`. The original config was exactly as it is here only minus the `branchConcurrentLimit`.

Even after deleting all the branches, disabling and re-enabling Renovate, I'm seeing tons of branches (one for proposed update) getting opened.

I am seeing the dependency dashboard show up in issues; could that somehow be related?

Current version of the config:
```json
{
  "extends": ["config:base"],
  "hostRules": [
    {
      "hostType": "npm",
      "matchHost": "registry.npmjs.org",
      "encrypted": {
        "token": "xyz"
      }
    }
  ],
  "branchConcurrentLimit": 5,
  "prConcurrentLimit": 5,
  "semanticCommits": true
}
```

even with this setting, the robot army is swarming 🤖

**Comment by wyardley** [comment]
also, `renovate-config-validator` seemed to accept both `5` and `"5"` for the integer values - not sure if the latter is valid?

**Comment by HonkingGoose** [comment]
> I am seeing the dependency dashboard show up in issues; could that somehow be related?

Version `v26` of Renovate defaults to have the Dependency Dashboard enabled if you extend from `config:base` like you do. This shouldn't be causing any issues. 😉

**Comment by rarkins** [comment]
I created this repo and installed the app and the rate limit seems correct: https://github.com/renovate-tests/npm63/issues/2

![image](https://user-images.githubusercontent.com/6311784/131692729-dcacc1d4-c967-48c4-8708-29c9988dba4b.png)

**Comment by wyardley** [comment]
Thanks @rarkins

So, I have that "rate limited" text as well, however, there seem to be a good 65 items on that list, and I seem to have a branch for each one of them, with our without the explicit branch concurrency limit in the config.

![image](https://user-images.githubusercontent.com/10714426/131726026-481adb34-7cb3-4d71-a6cd-7ac69d555444.png)

The config otherwise should be valid, right?

**Reply by rarkins** [reply]
My example repo has four branches like you'd expect:

![image](https://user-images.githubusercontent.com/6311784/131728677-77a7d76d-0e01-4dd6-b322-783d3d3c0a71.png)

If you added the limits *after* the branches existed then they won't prune any existing branches.

It's entirely possible there was a bug at some point, but I'm not able to reproduce it and can't do anything about it without some logs when the branches were created.

**Comment by wyardley** [comment]
I disabled the GH app, manually deleted the branches on the remote origin (once in web UI, a different time via CLI); on re- enabling the GH app for the repo, they start getting recreated immediately (and predictably). Is there anything else I can do to help reset that? Do I need to delete the GH tracking issue first or anything else?

I can send you the repo name privately if it helps at all.

**Reply by rarkins** [reply]
What do the new logs say about the limits?

**Reply by wyardley** [reply]
This is whitesource-hosted, so I don't know if I can get to the logs?

**Reply by rarkins** [reply]
app.renovatebot.com

**Reply by wyardley** [reply]
Amazing! Clearly it's been a while since I used this version of the product. Thanks for the pointer - will take a look.

**Comment by wyardley** [comment]
So, re-enabled things today, and looking good so far. I couldn't find anything obvious in the logs from yesterday about what happened, but will look into that if it happens again.

**Comment by les2** [comment]
I'm seeing the same behavior.

I have an org-wide shared config under `orgname/renovate-config` with a `default.json`:

`// default.json`
```
{
  "extends": [
    "config:base",
    ":preserveSemverRanges",
    "github>theRealDecile/renovate-config:ignoreInternalPythonPackages",
    "github>theRealDecile/renovate-config:ignoreInternalNpmPackages",
    "github>theRealDecile/renovate-config:npmArtifactory",
    "github>theRealDecile/renovate-config:limitConcurrentPrs"
  ]
}
```

The `limitConcurrentPrs.json` file contains:
```
{
  "prConcurrentLimit": 10
}
```

In order to force the onboarding PR to be recreated, I:
- renamed the "Configure Renovate" PR title (changed it to "closing to trigger onboarding PR")
- closed PR
- deleted branch

Then renovate opened a new onboarding PR like this:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>theRealDecile/renovate-config"
  ]
}
```

The onboarding PR description contains this:

```
Configuration Summary
Based on the default config's presets, Renovate will:

...
- Rate limit PR creation to a maximum of two per hour
- Limit to maximum 20 open PRs at any time
...
```

<img width="781" alt="Screen Shot 2021-09-09 at 19 17 27" src="https://user-images.githubusercontent.com/1066080/132749051-96b21163-62dc-44ca-992d-650555d89b31.png">

**Reply by rarkins** [reply]
@les2 the description you're seeing is autogenerated from `description` fields in the presets you extend. So you're seeing the description taken from `config:base`. If you were to add `"ignorePresets": [":prConcurrentLimit20"]` to your own preset then you should see that line disappear.
